src/server/components/dashboard/controllers.ts

```diff
@@ -349,3 +365,114 @@ export async function feedbackController(
     next(error);
   }
 }
+
+export async function listsEditAnnualReviewDateController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+    const { listId } = req.params;
+    const list = await findListById(listId);
+    const annualReviewStartDate = formatAnnualReviewDate((list as List), "annualReviewStartDate");
+
+    res.render("dashboard/lists-edit-annual-review-date", {
+      ...DEFAULT_VIEW_PROPS,
+      error: {
+        text: req.flash('annualReviewError')[0],
+        href: "#annualReviewDateForm",
+      },
+      annualReviewStartDate,
+      list,
+      csrfToken: getCSRFToken(req),
+    });
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDateController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+export async function listsEditAnnualReviewDatePostController(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+
+    return (req.body.action === "confirmNewDate")
+      ? await confirmNewAnnualReviewDate(req, res)
+      : await updateNewAnnualReviewDate(req, res);
+
+  } catch(error) {
+    logger.error(`listsEditAnnualReviewDatePostController Error: ${(error as Error).message}`)
+    next(error);
+  }
+}
+
+async function confirmNewAnnualReviewDate(req: Request, res: Response): Promise<void> {
+  const todaysDate = new Date();
```

Comment:
feels like this date validation / manipulation should be wrapped up in it's own helper function

Reply:
it should also have some unit tests to test the logic as well so can you include these:

Testing same year for current and new annual review
- current date in Jan, new date in Feb (within 6 months)
- current date in Jan, new date in Aug (longer than 6 months)
- current date in Jan, new date in Feb 29th leap year

Testing different year for current and new annual review
- current date in Nov, new date in Dec (within 6 months, same year)
- current date in Nov, new date in Jan (within 6 months, different years)
- current date in Nov, new date in Aug (longer than 6 months)
- current date in Nov, new date in Feb 29th leap year (different years)